test/controllers/shared/basic_rest_response_test.rb

```diff
@@ -36,5 +36,110 @@ def basic_edit_test(object_found = nil)
         end
       end
     end
+
+    def basic_pagination_per_page_test
+      context 'GET #index' do
+        setup do
+          @entries_per_page = Setting[:entries_per_page] ? Setting[:entries_per_page] : 5
+          factory_name =  get_factory_name(@controller.resource_class)
```

Comment:
Since you always pass the same parameter which is based on the instance variable which is also available to the function, there is no real need to pass it as a parameter and save the result etc. If you access the controller directly inside `get_factory_name` you can just use `create_model_list(get_factory_name, @entries_per_page + 2, *@factory_options)`